### PR TITLE
[EASI-3738] FE work for geography questions updates

### PR DESCRIPTION
**Pull request by garyjzhao**
# EASI-3738

## Changes and Description

- Updated General Characteristics read only section 
- Update `TargetsAndOPtions` forms 
- Add new `geographiesStatesAndTerritories` and `geographiesRegionTypes` to apolloGQL files
- Add new `geographiesStatesAndTerritories` and `geographiesRegionTypes` translation files
- Change 2

## How to test this change

1. Go to Page 4 in General Characteristics
2. Fill out Targeted Geography question
3. Ensure data persist, is in csv, and read only

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

**Comment by StevenWadeOddball** [comment]
@garyjzhao, is this just meant to be front end changes? Or front end and backend changes?

File: src/data/mock/readonly.ts [hunk 1]
```diff
@@ -172,6 +172,8 @@ const generalCharacteristicData: GetAllGeneralCharacteristicsTypes = {
   communityPartnersInvolvedNote: 'frwegqergqgrqwg planContractUpdatedNote',
   geographiesTargeted: true,
   geographiesTargetedTypes: [GeographyType.OTHER],
+  geographiesStatesAndTerritories: [],
+  geographiesRegionTypes: [],
```
Comment (patrickseguraoddball) [inline src/data/mock/readonly.ts:176]: Not a huge deal, but for these readonly tests, it's useful to always try to populate data for the snapshots

File: src/views/ModelPlan/TaskList/GeneralCharacteristics/TargetsAndOptions/index.tsx [hunk 2]
```diff
@@ -254,179 +262,210 @@ const TargetsAndOptions = () => {
...
+                                        }
+                                      </FieldErrorMsg>
+                                      <Field
+                                        as={TextAreaField}
+                                        data-testid="plan-characteristics-geographies-targeted-other"
+                                        id="plan-characteristics-geographies-targeted-other"
+                                        maxLength={5000}
+                                        className="mint-textarea"
+                                        name="geographiesTargetedTypesOther"
+                                      />
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/GeneralCharacteristics/TargetsAndOptions/index.tsx:390]: This should be a TextInput and not a TextAreaField

File: src/views/ModelPlan/TaskList/GeneralCharacteristics/TargetsAndOptions/index.tsx [hunk 3]
```diff
@@ -254,179 +262,210 @@ const TargetsAndOptions = () => {
...
+                        <FieldGroup
+                          scrollElement="geographiesTargetedTypes"
+                          error={!!flatErrors.geographiesTargetedTypes}
+                          className="margin-top-4"
+                        >
+                          <Label htmlFor="geographiesTargetedTypes">
+                            {generalCharacteristicsT(
+                              'geographiesTargetedTypes.label'
+                            )}
+                          </Label>
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/GeneralCharacteristics/TargetsAndOptions/index.tsx:274]: This label should not be bold

File: src/views/ModelPlan/TaskList/GeneralCharacteristics/TargetsAndOptions/index.tsx [hunk 4]
```diff
@@ -254,179 +262,210 @@ const TargetsAndOptions = () => {
...
+                        <FieldGroup
+                          scrollElement="geographiesTargetedAppliedTo"
+                          error={!!flatErrors.geographiesTargetedAppliedTo}
+                          className="margin-top-4"
+                        >
+                          <Label htmlFor="geographiesTargetedAppliedTo">
+                            {generalCharacteristicsT(
+                              'geographiesTargetedAppliedTo.label'
+                            )}
+                          </Label>
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/GeneralCharacteristics/TargetsAndOptions/index.tsx:407]: This label should not be bold

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
Nice work Gary

The new data needs to be added to both queries and file for CSV.  Currently `geographiesStatesAndTerritories` and `geographiesRegionTypes` are missing

**Comment by garyjzhao** [comment]
Hi @StevenWadeOddball, yes, this is only the FE changes. It may have tagged BE because it had branched off work @OddTomBrooks did on the BE. Please disregard BE review in this ticket

**Comment by StevenWadeOddball** [comment]
Thanks @garyjzhao , I'll hold off on review of this then. To note, if you merge this to the feature branch before the BE branch, this will show incorrect git history (because we use squash and merge). I"ll take myself off as a reviewer

**Comment by garyjzhao** [comment]
@StevenWadeOddball thanks for the info. I put on a `do not merge` tag until the BE is ready to go

**Comment by patrickseguraoddball** [comment]
Also the csv label header isnt rendering correctly, nor the data being populated.  i had populated some states here

Screenshot 2024-01-18 at 10 51 45 AM

File: src/views/ModelPlan/ReadOnly/GeneralCharacteristics/index.tsx [hunk 5]
```diff
@@ -483,15 +483,23 @@ const ReadOnlyGeneralCharacteristics = ({
...
+                          {generalCharacteristicsT(
+                            `geographiesRegionTypes.options.${region}`
+                          )}
+                        </li>
+                      );
+                    })
+                  ) : (
+                    <em className="text-base">
+                      {generalCharacteristicsMiscT('noAdditionalInformation')}
+                    </em>
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/ReadOnly/GeneralCharacteristics/index.tsx:500]: You can just pass null here if there are no values selected and the component will render `No additional information specified` correctly

File: src/i18n/en-US/modelPlan/generalCharacteristics.ts [hunk 6]
```diff
@@ -658,7 +658,8 @@ export const generalCharacteristicsMisc: Record<string, string> = {
   heading: 'General characteristics',
   clearanceHeading: 'Review general characteristics',
   breadcrumb: 'General characteristics',
-  specificQuestions: 'Key characteristic specific questions'
+  specificQuestions: 'Key characteristic specific questions',
+  noAdditionalInformation: 'No additional information specified'
```
Comment (patrickseguraoddball) [inline src/i18n/en-US/modelPlan/generalCharacteristics.ts:662]: Per the other comment, this translation isn't needed

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]

**Review by patrickseguraoddball — APPROVED** [review]
LG!

**Pull request merged**